File: commonlib/src/main/java/com/hjl/commonlib/utils/SpUtils.java
package com.hjl.commonlib.utils;

import android.content.Context;
import android.content.SharedPreferences;

import com.hjl.commonlib.base.BaseApplication;

/**
 * author: long
 * description please add a description here
 * Date: 2020/8/17
 */
public class SpUtils {

    private static final String COMMON_SP = "common_sp";

    private static final String SP_COOKIE = "sp_cookie";

    public static void clearCookie(){
        addCookie("");
    }

    public static void addCookie(String cookie){
        SharedPreferences sharedPreferences = getCommonSP();
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.putString(SP_COOKIE,cookie);
        edit.apply();
    }

    public static String getCookie(){
        return getCommonSP().getString(SP_COOKIE,"");
    }

    public static SharedPreferences getCommonSP(){
        return BaseApplication.getApplication().getSharedPreferences(COMMON_SP, Context.MODE_PRIVATE);
    }

    public static String getData(String key){
        return getCommonSP().getString(key,"");
    }

    public static void saveData(String key,String data){
        getCommonSP().edit().putString(key,data).apply();
    }

}

File: module_core/src/main/java/com/hjl/core/utils/SpUtils.kt
package com.hjl.core.utils

import android.content.Context
import android.content.SharedPreferences
import android.util.Base64
import com.hjl.commonlib.base.BaseApplication
import com.tencent.bugly.crashreport.biz.UserInfoBean
import com.hjl.core.net.bean.*


/**
 * author: long
 * description please add a description here
 * Date: 2020/8/17
 */
object SpUtils {

    val SP_TAG_USER = "sp_tag_user"

    val SP_USER_INFO = "sp_user_info"


    fun saveUserInfo(userInfo: String){
        val editor = getUserInfoSP().edit()
        editor.putString(SP_USER_INFO, Base64.encodeToString(userInfo.toByteArray(),Base64.NO_PADDING))
        editor.apply()
    }

    fun getUserInfo():String{
        return String(Base64.decode(getUserInfoSP().getString(SP_USER_INFO,""),Base64.NO_PADDING))
    }

    private fun getUserInfoSP():SharedPreferences{
        return BaseApplication.getApplication().getSharedPreferences(SP_TAG_USER,Context.MODE_PRIVATE)
    }


}